debezium-ddl-parser/src/main/java/io/debezium/antlr/mysql/MySqlAntlrDdlParser.java

```diff
@@ -0,0 +1,618 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.antlr.mysql;
+
+import io.debezium.antlr.AntlrDdlParser;
+import io.debezium.antlr.ProxyParseTreeListener;
+import io.debezium.ddl.parser.mysql.generated.MySqlLexer;
+import io.debezium.ddl.parser.mysql.generated.MySqlParser;
+import io.debezium.ddl.parser.mysql.generated.MySqlParserBaseListener;
+import io.debezium.relational.Column;
+import io.debezium.relational.ColumnEditor;
+import io.debezium.relational.Table;
+import io.debezium.relational.TableEditor;
+import io.debezium.relational.TableId;
+import io.debezium.text.ParsingException;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.misc.Interval;
+import org.antlr.v4.runtime.tree.ParseTree;
+
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author Roman Kuchár <kucharrom@gmail.com>.
+ */
+public class MySqlAntlrDdlParser extends AntlrDdlParser<MySqlLexer, MySqlParser> {
+
+    private TableEditor tableEditor;
+    private ColumnEditor columnEditor;
+
+    @Override
+    protected ParseTree parseTree(MySqlParser parser) {
+        return parser.root();
+    }
+
+    @Override
+    protected void assignParserListeners(ProxyParseTreeListener proxyParseTreeListener) {
+        proxyParseTreeListener.add(new CreateTableParserListener());
+        proxyParseTreeListener.add(new DropTableParserListener());
+        proxyParseTreeListener.add(new AlterTableParserListener());
+        proxyParseTreeListener.add(new ColumnDefinitionParserListener());
```

Comment:
I'm suprised to see this as a separate listener. How does this manage state together with let's say the create table listener?